# sckd
A comparative learning and knowledge distillation based approach to scRNA-seq data analysis

Single-cell heterogeneity analyses face challenges of high dimensionality, complexity, and noise, particularly in the context of precise cell type classification. Although various analytical approaches have been utilized for scRNA-seq data analysis, they often underperform in diverse biological applications, mainly due to limitations in generalisation ability and adaptability, which may lead to bias in identifying cell subpopulations, and consequently affecting the understanding of diseases, therapeutic responses and biological processes. To tackle these challenges, this study introduces a scRNA-seq data analysis method called scKD, which combines a hybrid neighbourhood-enhanced comparative learning model and a self-knowledge distiller model strategy aimed at enhancing the clustering accuracy of single-cell transcriptomic data. In addition to accurately detecting major cell types, the method is also capable of efficiently recognizing rare cell subtypes, offering new insights into the study of cellular heterogeneity. In this study, extensive experiments were conducted on a variety of real datasets, and the experimental results demonstrated that scKD showed good performance in cell subpopulation identification, clustering stability, and robustness, proving that scKD is a highly effective and dependable approach for analyzing single-cell RNA transcriptomic data, which provides strong support for a deeper insight into cellular heterogeneity.
